File: superstitioApi/java/superstitioapi/relicToBlight/customSave/InfoBlightConstructSaveFilePatch.java
package superstitioapi.relicToBlight.customSave;

import basemod.abstracts.CustomSavableRaw;
import com.evacipated.cardcrawl.modthespire.lib.SpirePatch;
import com.evacipated.cardcrawl.modthespire.lib.SpirePostfixPatch;
import com.megacrit.cardcrawl.blights.AbstractBlight;
import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
import com.megacrit.cardcrawl.saveAndContinue.SaveFile;
import superstitioapi.relicToBlight.BlightWithRelic;

import static superstitioapi.relicToBlight.customSave.SuperstitioApiModSaves.ArrayListOfJsonElement;
import static superstitioapi.relicToBlight.customSave.SuperstitioApiModSaves.modBlightSaves;

@SpirePatch(clz = SaveFile.class, method = "<ctor>", paramtypez = {SaveFile.SaveType.class})
public class InfoBlightConstructSaveFilePatch {
    @SpirePostfixPatch
    public static void Prefix(final SaveFile __instance, final SaveFile.SaveType saveType) {
        final ArrayListOfJsonElement blightSaves = new ArrayListOfJsonElement();
        for (AbstractBlight blight : AbstractDungeon.player.blights) {
            if (blight instanceof BlightWithRelic
                    && ((BlightWithRelic) blight).relic instanceof CustomSavableRaw)
                blightSaves.add(((CustomSavableRaw) ((BlightWithRelic) blight).relic).onSaveRaw());
            else {
                blightSaves.add(null);
            }
        }
        modBlightSaves.set(__instance, blightSaves);
    }
}

File: superstitioApi/java/superstitioapi/relicToBlight/customSave/InfoBlightConstructSaveFilePatch.kt
package superstitioapi.relicToBlight.customSave

import basemod.abstracts.CustomSavableRaw
import com.evacipated.cardcrawl.modthespire.lib.SpirePatch2
import com.evacipated.cardcrawl.modthespire.lib.SpirePostfixPatch
import com.megacrit.cardcrawl.dungeons.AbstractDungeon
import com.megacrit.cardcrawl.saveAndContinue.SaveFile
import com.megacrit.cardcrawl.saveAndContinue.SaveFile.SaveType
import superstitioapi.relicToBlight.BlightWithRelic

@SpirePatch2(clz = SaveFile::class, method = "<ctor>", paramtypez = [SaveType::class])
object InfoBlightConstructSaveFilePatch
{
    @SpirePostfixPatch
    @JvmStatic
    fun Prefix(__instance: SaveFile?, type: SaveType?)
    {
        val blightSaves = SuperstitioApiModSaves.ArrayListOfJsonElement()
        for (blight in AbstractDungeon.player.blights)
        {
            if (blight is BlightWithRelic && blight.relic is CustomSavableRaw)
                blightSaves.add((blight.relic as CustomSavableRaw).onSaveRaw())
            else
            {
                blightSaves.add(null)
            }
        }
        SuperstitioApiModSaves.modBlightSaves.set(__instance, blightSaves)
    }
}
